<?xml version="1.0"?>
<launch>
    
    <arg name="arg_anquan_juli" value="4.5"/> <!-- 安全距离 -->
    <arg name="arg_xishu_linearx" value="0.5"/> <!-- 线速度PID的P -->
    <arg name="arg_xishu_angularz" value="2"/> <!-- 角速度PID的P -->
    <arg name="arg_yuzhi_x" value="4"/> <!-- 线速度上限 -->
    <arg name="arg_yuzhi_z" value="3"/> <!-- 角速度上限 -->
    

    <!-- 目标跟踪 -->
    <node pkg="multiple_target_tracking" type="multiple_target_tracking_node" name="multiple_target_tracking_node" output="screen">
        <param name="sub_topic" value="/text_det"/>
        <param name="pub_topic" value="/objects_tracked"/>
        <param name="pub_topic_obstacle_array" value="/obstacle_array"/>
        <param name="pub_topic_position" value="/obj_xyz"/>
        <param name="frame_id" value="livox_frame"/> <!-- 需要根据当前frame调整 -->
        <param name="confidence" value="0.38"/> <!-- 可调整跟踪的置信度 -->
        <param name="corrupt_time" value="3.0"/> <!-- 可调整跟踪切换的时间，该时间内插入人不会切换跟踪 -->
        
        <param name="show_objects_num" value="true"/>
        <param name="show_time" value="false"/>

        <param name="time_interval" value="0.1"/>
        <!-- 跟踪的残差阈值，速度过快以至于跟踪不上时可能需要调整 -->
        <param name="gate_threshold" value="200"/>
        <param name="blind_update_limit" value="4"/>
        
        <param name="sigma_ax" value="1"/>
        <param name="sigma_ay" value="1"/>
        <param name="sigma_ox" value="0.1"/>
        <param name="sigma_oy" value="0.1"/>
        
        <!-- 丢弃超过该尺寸的目标，已不再需要调整 -->
        <param name="min_scale" value="0.2"/>
        <param name="max_scale" value="0.8"/>
        <param name="min_height" value="1.1"/>
        <param name="max_height" value="2.0"/>
    </node>

    <node pkg="follow" type="follow_node" name="follow_node" output="screen">
        <param name="yuzhi_anquan" value="$(arg arg_anquan_juli)"/>
        <param name="debug_on" value="1"/>
        <param name="xishu_linearx" value="0.5"/>
        <param name="xishu_angularz" value="2"/>
        <param name="yuzhi_x" value="4"/>
        <param name="yuzhi_z" value="3"/>
        <param name="info_print" value="false"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find multiple_target_tracking)/launch/track.rviz" />
</launch>
